modules/rostests/apitests/ntdll/LdrLoadDll.c

```diff
@@ -0,0 +1,100 @@
+    WCHAR *pch;
+
+    RtlInitEmptyUnicodeString(&DllName, NULL, 0);
+    GetWindowsDirectoryW(szWinDir, _countof(szWinDir));
+    GetSystemDirectoryW(szSysDir, _countof(szSysDir));
+
+    StartSeh() LdrLoadDll(NULL, NULL, &DllName, NULL);          EndSeh(STATUS_ACCESS_VIOLATION);
+    StartSeh() LdrLoadDll(NULL, NULL, &DllName, &BaseAddress);  EndSeh(STATUS_ACCESS_VIOLATION);
+    StartSeh() LdrLoadDll(L"", NULL, &DllName, NULL);           EndSeh(STATUS_ACCESS_VIOLATION);
+    StartSeh() LdrLoadDll(L"", NULL, &DllName, &BaseAddress);   EndSeh(STATUS_ACCESS_VIOLATION);
+
+    RtlInitUnicodeString(&DllName, L"advapi32.dll");
+    StartSeh()
+        Status = LdrLoadDll(NULL, NULL, &DllName, NULL);
+        if (NT_SUCCESS(Status)) LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+
+    RtlInitUnicodeString(&DllName, L"advapi32.dll");
+    StartSeh()
+        BaseAddress = InvalidPointer;
+        Status = LdrLoadDll(NULL, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        ok(BaseAddress != NULL && BaseAddress != InvalidPointer, "BaseAddress = %p\n", BaseAddress);
+        if (NT_SUCCESS(Status))
+        {
+            BaseAddress2 = InvalidPointer;
+            Status = LdrLoadDll(NULL, NULL, &DllName, &BaseAddress2);
+            ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+            ok(BaseAddress2 == BaseAddress, "BaseAddress2 = %p, expected %p\n", BaseAddress2, BaseAddress);
+            LdrUnloadDll(BaseAddress);
+        }
+    EndSeh(STATUS_SUCCESS);
+
+    RtlInitUnicodeString(&DllName, L"advapi32.dll");
+    StartSeh()
+        Status = LdrLoadDll(L"\\SystemRoot\\System32", NULL, &DllName, NULL);
+    EndSeh(STATUS_ACCESS_VIOLATION);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32", szSysDir);
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status)) LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s/advapi32", szSysDir);
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status)) LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
+
+    StringCchPrintfW(szPath, _countof(szPath), L"%s\\advapi32", szSysDir);
+
+    for (pch = szPath; *pch != UNICODE_NULL; ++pch)
+    {
+        if (*pch == L'\\')
+            *pch = L'/';
+    }
+
+    RtlInitUnicodeString(&DllName, szPath);
+    StartSeh()
+        Status = LdrLoadDll(szWinDir, NULL, &DllName, &BaseAddress);
+        ok(Status == STATUS_SUCCESS, "Status = %lx\n", Status);
+        if (NT_SUCCESS(Status)) LdrUnloadDll(BaseAddress);
+    EndSeh(STATUS_SUCCESS);
```

Comment:
I would like that you also test dll name _**with the `.dll` extension appended**_, because it seems that the LdrpLoadDll code path you have touched is also sensitive to that.

Reply:
Fixed.